simd/src/Kokkos_SIMD_AVX2.hpp

```diff
@@ -680,10 +680,12 @@ template <>
 class simd<std::int64_t, simd_abi::avx2_fixed_size<4>> {
   __m256i m_value;
 
+  static_assert(sizeof(long long) == 8);
+
  public:
   using value_type = std::int64_t;
   using abi_type   = simd_abi::avx2_fixed_size<4>;
-  using mask_type  = simd_mask<value_type, abi_type>;
+  using mask_type  = simd_mask<std::int64_t, abi_type>;
```

Comment:
Is this change necessary, seeing that `value_type` was already `std::int64_t` from `using value_type = std::int64_t`?

Reply:
No, that was for another attempt. An alternative is
```
diff --git a/simd/src/Kokkos_SIMD_AVX2.hpp b/simd/src/Kokkos_SIMD_AVX2.hpp
index 80f96153d..ad4f53625 100644
--- a/simd/src/Kokkos_SIMD_AVX2.hpp
+++ b/simd/src/Kokkos_SIMD_AVX2.hpp
@@ -683,7 +683,7 @@ class simd<std::int64_t, simd_abi::avx2_fixed_size<4>> {
   static_assert(sizeof(long long) == 8);
 
  public:
-  using value_type = std::int64_t;
+  using value_type = long long;
   using abi_type   = simd_abi::avx2_fixed_size<4>;
   using mask_type  = simd_mask<std::int64_t, abi_type>;
   using reference  = value_type&;
@@ -729,12 +729,12 @@ class simd<std::int64_t, simd_abi::avx2_fixed_size<4>> {
   }
   KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION void copy_from(value_type const* ptr,
                                                        element_aligned_tag) {
-    m_value = _mm256_maskload_epi64(reinterpret_cast<long long const*>(ptr),
+    m_value = _mm256_maskload_epi64(ptr, static_cast<__m256i>(mask_type(true)));
   }
   KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION void copy_to(
       value_type* ptr, element_aligned_tag) const {
-    _mm256_maskstore_epi64(reinterpret_cast<long long*>(ptr),
+    _mm256_maskstore_epi64(ptr, static_cast<__m256i>(mask_type(true)), m_value);
   }
   KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION constexpr explicit operator __m256i()
```
avoiding the possible undefined behavior due to `reinterpret_cast`. Of course, this would change the value type and it's not quite clear to me if that's OK/better or not. Making `long long` a storage type or similar might be another option.

Reply:
I think setting `value_type` to be `long long` is something that could revisited. If all compilers' implementation of all avx2 intrinsics consistently use `long long`, then it may be better to set `value_type` to be `long long`. But for now, I think we can proceed with having a `reinterpret_cast` here.